File: my-solutions/13. Roman to Integer/code.py
# method 1
class Solution:
    def romanToInt(self, s: str) -> int:
        num = 0

        sym_val = {
            "I" : 1,
            "V" : 5,
            "X" : 10,
            "L" : 50,
            "C" : 100,
            "D" : 500,
            "M" : 1000,
        }

        if "IV" in s:
            s = s.replace("IV", "")
            num += 4
        if "IX" in s:
            s = s.replace("IX", "")
            num += 9
        if "XL" in s:
            s = s.replace("XL", "")
            num += 40
        if "XC" in s:
            s = s.replace("XC", "")
            num += 90
        if "CD" in s:
            s = s.replace("CD", "")
            num += 400
        if "CM" in s:
            s = s.replace("CM", "")
            num += 900

        for l in s:
            num += sym_val[l]

        return num

# method 2
class Solution:
    def romanToInt(self, s: str) -> int:
        num = 0

        sym_val = {
            "I" : 1,
            "V" : 5,
            "X" : 10,
            "L" : 50,
            "C" : 100,
            "D" : 500,
            "M" : 1000,
        }

        for i, letter in enumerate(s):
            if i < len(s) - 1 and sym_val[s[i]] < sym_val[s[i + 1]] and i < len(s) - 1:
                    num -= sym_val[letter]
            else:
                num += sym_val[letter]

        return num

File: my-solutions/13. Roman to Integer/code.cs
public class Solution {
    public int RomanToInt(string s) {
        int num = 0;

        s = s.Replace("IV", "a");
        s = s.Replace("IX", "b");
        s = s.Replace("XL", "c");
        s = s.Replace("XC", "d");
        s = s.Replace("CD", "e");
        s = s.Replace("CM", "f");

        Dictionary<char, int> dict_1 = new Dictionary<char, int>();
        dict_1.Add('I', 1);
        dict_1.Add('V', 5);
        dict_1.Add('X', 10);
        dict_1.Add('L', 50);
        dict_1.Add('C', 100);
        dict_1.Add('D', 500);
        dict_1.Add('M', 1000);
        dict_1.Add('a', 4);
        dict_1.Add('b', 9);
        dict_1.Add('c', 40);
        dict_1.Add('d', 90);
        dict_1.Add('e', 400);
        dict_1.Add('f', 900);

        foreach(char letter in s)
        {
            num += dict_1[letter];
        }

        return num;
    }
}
